### ISSUE TITLE
[AARCH64] Some Binary primitives failing benchdnn with certain configs

### ISSUE THREAD

**Issue by AhSinan** [oneapi-src/oneDNN#1541]
# Summary
Hello,
We are noticing some failures while running benchdnn tests, specifically, **Binary ops**, with these configurations:
- `-stag=aBx8b:any` and `--inplace=true`. 
- It is affecting Neoverse-V1 (SVE-256) and A64fx (SVE-512) targets.

It seems this issue has started occurring after a recent patch, https://github.com/oneapi-src/oneDNN/commit/7ef0b2ef2255d49ee5e334954ac8e0d2d3853ac5

# Version
```
onednn_verbose,info,oneDNN v3.0.0 (commit 5e237df1e9049c2e3e9102cab1785454db00cf8b)
onednn_verbose,info,cpu,runtime:OpenMP,nthr:32
onednn_verbose,info,cpu,isa:AArch64 SVE (256 bits)
```

# Environment
```
Architecture:                    aarch64
CPU op-mode(s):                  32-bit, 64-bit
Byte Order:                      Little Endian
CPU(s):                          32
On-line CPU(s) list:             0-31
Thread(s) per core:              1
Core(s) per socket:              32
Socket(s):                       1
NUMA node(s):                    1
Vendor ID:                       ARM
Model:                           1
Stepping:                        r1p1
BogoMIPS:                        2100.00
L1d cache:                       2 MiB
L1i cache:                       2 MiB
L2 cache:                        32 MiB
L3 cache:                        32 MiB
NUMA node0 CPU(s):               0-31
Vulnerability Itlb multihit:     Not affected
Vulnerability L1tf:              Not affected
Vulnerability Mds:               Not affected
Vulnerability Meltdown:          Not affected
Vulnerability Mmio stale data:   Not affected
Vulnerability Retbleed:          Not affected
Vulnerability Spec store bypass: Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:        Mitigation; __user pointer sanitization
Vulnerability Spectre v2:        Mitigation; CSV2, BHB
Vulnerability Srbds:             Not affected
Vulnerability Tsx async abort:   Not affected
Flags:                           fp asimd evtstrm aes pmull sha1 sha2 crc32 atomics fphp asimdhp cpuid asimdrdm jscvt fcma lrcpc dcpop sha3 sm3 sm4 asimddp sha512 sve asimdfhm dit uscat ilrcpc flagm ssbs paca pa
                                 cg dcpodp svei8mm svebf16 i8mm bf16 dgh rng
```

* OS version 
```
uname -a
Linux #32~20.04.1-Ubuntu SMP Mon Jan 9 18:02:12 UTC 2023 aarch64 aarch64 aarch64 GNU/Linux
```

* Compiler version (`gcc --version`)
```
gcc-10 --version
gcc-10 (Ubuntu 10.3.0-1ubuntu1~20.04) 10.3.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

```

* CMake version (`cmake --version`)
```
cmake version 3.16.3
CMake suite maintained and supported by Kitware (kitware.com/cmake).
```
* git hash (`git log -1 --format=%H`)
5e237df1e9049c2e3e9102cab1785454db00cf8b
# Steps to reproduce
Build was done with ACL backend:
ACL build with:
```
scons -j56 Werror=0 debug=0 neon=1 opencl=0  examples=0 embed_kernels=0 os=linux arch=arm64-v8.2-a build=native
```
oneDNN built with:
```
ACL_ROOT_DIR=/PATH/TO/ACL cmake .. -DCMAKE_BUILD_TYPE=Release -DDNNL_AARCH64_USE_ACL=1 -DDNNL_BUILD_FOR_CI=ON
make -j
```

These are the failing tests:
```
--binary --stag=aBx8b:any --alg=ge --inplace=true 1x1x1:1x1x1
--binary --stag=aBx8b:any --alg=gt --inplace=true 1x1x1:1x1x1
--binary --stag=aBx8b:any --alg=le --inplace=true 1x1x1:1x1x1
--binary --stag=aBx8b:any --alg=lt --inplace=true 1x1x1:1x1x1
--binary --stag=aBx8b:any --alg=eq --inplace=true 1x1x1:1x1x1
--binary --stag=aBx8b:any --alg=ne --inplace=true 1x1x1:1x1x1
--binary --stag=aBx8b:any --alg=ge --inplace=true 2x12x3:2x12x3
--binary --stag=aBx8b:any --alg=gt --inplace=true 2x12x3:2x12x3
--binary --stag=aBx8b:any --alg=le --inplace=true 2x12x3:2x12x3
--binary --stag=aBx8b:any --alg=lt --inplace=true 2x12x3:2x12x3
--binary --stag=aBx8b:any --alg=eq --inplace=true 2x12x3:2x12x3
--binary --stag=aBx8b:any --alg=ne --inplace=true 2x12x3:2x12x3
--binary --stag=aBx8b:any --alg=ge --inplace=true 2x44x3:2x44x3
--binary --stag=aBx8b:any --alg=gt --inplace=true 2x44x3:2x44x3
--binary --stag=aBx8b:any --alg=le --inplace=true 2x44x3:2x44x3
--binary --stag=aBx8b:any --alg=lt --inplace=true 2x44x3:2x44x3
--binary --stag=aBx8b:any --alg=eq --inplace=true 2x44x3:2x44x3
--binary --stag=aBx8b:any --alg=ne --inplace=true 2x44x3:2x44x3
--binary --stag=aBx8b:any --alg=ge --inplace=true 2x12x3x4:2x12x3x4
--binary --stag=aBx8b:any --alg=gt --inplace=true 2x12x3x4:2x12x3x4
--binary --stag=aBx8b:any --alg=le --inplace=true 2x12x3x4:2x12x3x4
--binary --stag=aBx8b:any --alg=lt --inplace=true 2x12x3x4:2x12x3x4
--binary --stag=aBx8b:any --alg=eq --inplace=true 2x12x3x4:2x12x3x4
--binary --stag=aBx8b:any --alg=ne --inplace=true 2x12x3x4:2x12x3x4
--binary --stag=aBx8b:any --alg=ge --inplace=true 2x44x3x4:2x44x3x4
--binary --stag=aBx8b:any --alg=gt --inplace=true 2x44x3x4:2x44x3x4
--binary --stag=aBx8b:any --alg=le --inplace=true 2x44x3x4:2x44x3x4
--binary --stag=aBx8b:any --alg=lt --inplace=true 2x44x3x4:2x44x3x4
--binary --stag=aBx8b:any --alg=eq --inplace=true 2x44x3x4:2x44x3x4
--binary --stag=aBx8b:any --alg=ne --inplace=true 2x44x3x4:2x44x3x4
--binary --stag=aBx8b:any --alg=ge --inplace=true 8x15x5x7:1x15x1x1
--binary --stag=aBx8b:any --alg=gt --inplace=true 8x15x5x7:1x15x1x1
--binary --stag=aBx8b:any --alg=le --inplace=true 8x15x5x7:1x15x1x1
--binary --stag=aBx8b:any --alg=lt --inplace=true 8x15x5x7:1x15x1x1
--binary --stag=aBx8b:any --alg=eq --inplace=true 8x15x5x7:1x15x1x1
--binary --stag=aBx8b:any --alg=ne --inplace=true 8x15x5x7:1x15x1x1
--binary --stag=aBx8b:any --inplace=true 8x15x5x7:8x1x5x7
--binary --stag=aBx8b:any --alg=mul --inplace=true 8x15x5x7:8x1x5x7
--binary --stag=aBx8b:any --alg=max --inplace=true 8x15x5x7:8x1x5x7
--binary --stag=aBx8b:any --alg=min --inplace=true 8x15x5x7:8x1x5x7
--binary --stag=aBx8b:any --alg=sub --inplace=true 8x15x5x7:8x1x5x7
--binary --stag=aBx8b:any --alg=ge --inplace=true 8x15x5x7:8x1x5x7
--binary --stag=aBx8b:any --alg=gt --inplace=true 8x15x5x7:8x1x5x7
--binary --stag=aBx8b:any --alg=le --inplace=true 8x15x5x7:8x1x5x7
--binary --stag=aBx8b:any --alg=lt --inplace=true 8x15x5x7:8x1x5x7
--binary --stag=aBx8b:any --alg=eq --inplace=true 8x15x5x7:8x1x5x7
--binary --stag=aBx8b:any --alg=ne --inplace=true 8x15x5x7:8x1x5x7
--binary --stag=aBx8b:any --inplace=true 8x15x5x7:1x1x1x7
--binary --stag=aBx8b:any --alg=mul --inplace=true 8x15x5x7:1x1x1x7
--binary --stag=aBx8b:any --alg=max --inplace=true 8x15x5x7:1x1x1x7
--binary --stag=aBx8b:any --alg=min --inplace=true 8x15x5x7:1x1x1x7
--binary --stag=aBx8b:any --alg=sub --inplace=true 8x15x5x7:1x1x1x7
--binary --stag=aBx8b:any --alg=ge --inplace=true 8x15x5x7:1x1x1x7
--binary --stag=aBx8b:any --alg=gt --inplace=true 8x15x5x7:1x1x1x7
--binary --stag=aBx8b:any --alg=le --inplace=true 8x15x5x7:1x1x1x7
--binary --stag=aBx8b:any --alg=lt --inplace=true 8x15x5x7:1x1x1x7
--binary --stag=aBx8b:any --alg=eq --inplace=true 8x15x5x7:1x1x1x7
--binary --stag=aBx8b:any --alg=ne --inplace=true 8x15x5x7:1x1x1x7
--binary --stag=aBx8b:any --alg=ge --inplace=true 8x15x5x7:1x15x5x7
--binary --stag=aBx8b:any --alg=gt --inplace=true 8x15x5x7:1x15x5x7
--binary --stag=aBx8b:any --alg=le --inplace=true 8x15x5x7:1x15x5x7
--binary --stag=aBx8b:any --alg=lt --inplace=true 8x15x5x7:1x15x5x7
--binary --stag=aBx8b:any --alg=eq --inplace=true 8x15x5x7:1x15x5x7
--binary --stag=aBx8b:any --alg=ne --inplace=true 8x15x5x7:1x15x5x7
--binary --stag=aBx8b:any --inplace=true 8x15x5x7:1x1x1x1
--binary --stag=aBx8b:any --alg=mul --inplace=true 8x15x5x7:1x1x1x1
--binary --stag=aBx8b:any --alg=max --inplace=true 8x15x5x7:1x1x1x1
--binary --stag=aBx8b:any --alg=min --inplace=true 8x15x5x7:1x1x1x1
--binary --stag=aBx8b:any --alg=sub --inplace=true 8x15x5x7:1x1x1x1
--binary --stag=aBx8b:any --alg=ge --inplace=true 8x15x5x7:1x1x1x1
--binary --stag=aBx8b:any --alg=gt --inplace=true 8x15x5x7:1x1x1x1
--binary --stag=aBx8b:any --alg=le --inplace=true 8x15x5x7:1x1x1x1
--binary --stag=aBx8b:any --alg=lt --inplace=true 8x15x5x7:1x1x1x1
--binary --stag=aBx8b:any --alg=eq --inplace=true 8x15x5x7:1x1x1x1
--binary --stag=aBx8b:any --alg=ne --inplace=true 8x15x5x7:1x1x1x1
--binary --stag=aBx8b:any --inplace=true --attr-post-ops=add:f32:per_oc 2x12x3:2x12x3
--binary --stag=aBx8b:any --inplace=true --attr-post-ops=ge:f32 2x12x3:2x12x3
--binary --stag=aBx8b:any --inplace=true --attr-post-ops=add:f32:per_oc 2x44x3:2x44x3
--binary --stag=aBx8b:any --inplace=true --attr-post-ops=ge:f32 2x44x3:2x44x3
--binary --stag=aBx8b:any --inplace=true --attr-post-ops=add:f32:per_oc 2x12x3x4:2x12x3x4
--binary --stag=aBx8b:any --inplace=true --attr-post-ops=ge:f32 2x12x3x4:2x12x3x4
--binary --stag=aBx8b:any --inplace=true --attr-post-ops=add:f32:per_oc 2x44x3x4:2x44x3x4
--binary --stag=aBx8b:any --inplace=true --attr-post-ops=ge:f32 2x44x3x4:2x44x3x4
--binary --stag=aBx8b:any --inplace=true --attr-post-ops=add:f32:per_oc 8x15x5x7:1x15x1x1
--binary --stag=aBx8b:any --inplace=true --attr-post-ops=ge:f32 8x15x5x7:1x15x1x1
--binary --stag=aBx8b:any --inplace=true 8x15x5x7:8x1x5x7
--binary --stag=aBx8b:any --inplace=true --attr-post-ops=add:f32:per_oc 8x15x5x7:8x1x5x7
--binary --stag=aBx8b:any --inplace=true --attr-post-ops=ge:f32 8x15x5x7:8x1x5x7
--binary --stag=aBx8b:any --inplace=true 8x15x5x7:1x1x1x7
--binary --stag=aBx8b:any --inplace=true --attr-post-ops=add:f32:per_oc 8x15x5x7:1x1x1x7
--binary --stag=aBx8b:any --inplace=true --attr-post-ops=ge:f32 8x15x5x7:1x1x1x7
--binary --stag=aBx8b:any --inplace=true --attr-post-ops=add:f32:per_oc 8x15x5x7:1x15x5x7
--binary --stag=aBx8b:any --inplace=true --attr-post-ops=ge:f32 8x15x5x7:1x15x5x7
--binary --stag=aBx8b:any --inplace=true --attr-post-ops=mul:s8+add:f32+sum:0.5+abs 8x15x5x7:1x15x5x7
--binary --stag=aBx8b:any --inplace=true 8x15x5x7:1x1x1x1
--binary --stag=aBx8b:any --attr-post-ops=relu:-0.01+sum:2 8x15x5x7:1x1x1x1
--binary --stag=aBx8b:any --inplace=true --attr-post-ops=add:f32:per_oc 8x15x5x7:1x1x1x1
--binary --stag=aBx8b:any --inplace=true --attr-post-ops=ge:f32 8x15x5x7:1x1x1x1
--binary --stag=aBx8b:any --inplace=true 8x15x5x7:8x1x5x7
--binary --stag=aBx8b:any --inplace=true --attr-scales=src:common:0.125* 8x15x5x7:8x1x5x7
--binary --stag=aBx8b:any --inplace=true --attr-scales=src1:common:2* 8x15x5x7:8x1x5x7
--binary --stag=aBx8b:any --inplace=true --attr-scales=src:common:0.25*+src1:common:0.5* 8x15x5x7:8x1x5x7
--binary --stag=aBx8b:any --inplace=true 8x15x5x7:1x1x1x7
--binary --stag=aBx8b:any --inplace=true --attr-scales=src:common:0.125* 8x15x5x7:1x1x1x7
--binary --stag=aBx8b:any --inplace=true --attr-scales=src1:common:2* 8x15x5x7:1x1x1x7
--binary --stag=aBx8b:any --inplace=true --attr-scales=src:common:0.25*+src1:common:0.5* 8x15x5x7:1x1x1x7
--binary --stag=aBx8b:any --inplace=true 8x15x5x7:1x1x1x1
--binary --stag=aBx8b:any --inplace=true --attr-scales=src:common:0.125* 8x15x5x7:1x1x1x1
--binary --stag=aBx8b:any --inplace=true --attr-scales=src1:common:2* 8x15x5x7:1x1x1x1
--binary --stag=aBx8b:any --inplace=true --attr-scales=src:common:0.25*+src1:common:0.5* 8x15x5x7:1x1x1x1
```

# Observed behavior
<details>
  <summary>
    Sample output of these tests with <code>ONEDNN_VERBOSE=1</code>
  </summary>

```
onednn_verbose,info,oneDNN v3.0.0 (commit 5e237df1e9049c2e3e9102cab1785454db00cf8b)
onednn_verbose,info,cpu,runtime:OpenMP,nthr:32
onednn_verbose,info,cpu,isa:AArch64 SVE (256 bits)
onednn_verbose,info,gpu,runtime:none
onednn_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_f32::blocked:abcd:f0 dst_f32:p:blocked:aBcd8b:f0,,,1x1x1x1,0.00195312
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_f32::blocked:abcd:f0 dst_f32:p:blocked:aBcd8b:f0,,,8x15x5x7,0.408936
onednn_verbose,exec,cpu,binary,jit:uni,undef,src_f32:p:blocked:aBcd8b:f0 src_f32:p:blocked:aBcd8b:f0 dst_f32:p:blocked:aBcd8b:f0,,alg:binary_add,8x15x5x7:1x1x1x1,0.0222168
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_f32:p:blocked:aBcd8b:f0 dst_f32::blocked:abcd:f0,,,8x15x5x7,0.00390625
[ 280][DST][0:8:0:0] exp_f32:        -2.5 exp:        -2.5 got:         2.5 diff:       5 rdiff:       2
[ 281][DST][0:8:0:1] exp_f32:       -8.75 exp:       -8.75 got:       -3.75 diff:       5 rdiff:0.571429
[ 282][DST][0:8:0:2] exp_f32:         -15 exp:         -15 got:         -10 diff:       5 rdiff:0.333333
[ 283][DST][0:8:0:3] exp_f32:           0 exp:           0 got:           5 diff:       5 rdiff:       5
[ 284][DST][0:8:0:4] exp_f32:       -6.25 exp:       -6.25 got:       -1.25 diff:       5 rdiff:     0.8
[ 285][DST][0:8:0:5] exp_f32:       -12.5 exp:       -12.5 got:        -7.5 diff:       5 rdiff:     0.4
[ 286][DST][0:8:0:6] exp_f32:         2.5 exp:         2.5 got:         7.5 diff:       5 rdiff:       2
[ 287][DST][0:8:1:0] exp_f32:       -3.75 exp:       -3.75 got:        1.25 diff:       5 rdiff: 1.33333
[ 288][DST][0:8:1:1] exp_f32:         -10 exp:         -10 got:          -5 diff:       5 rdiff:     0.5
0:FAILED (errors:1960 total:4200) __REPRO: --binary --stag=aBx8b:any --inplace=true 8x15x5x7:1x1x1x1
tests:1 passed:0 skipped:0 mistrusted:0 unimplemented:0 invalid_arguments:0 failed:1 listed:0
total compute_ref: sum(s):0.00
onednn_verbose,info,oneDNN v3.0.0 (commit 5e237df1e9049c2e3e9102cab1785454db00cf8b)
onednn_verbose,info,cpu,runtime:OpenMP,nthr:32
onednn_verbose,info,cpu,isa:AArch64 SVE (256 bits)
onednn_verbose,info,gpu,runtime:none
onednn_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:a:f0 dst_f32::blocked:a:f0,,,1,0.581787
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_f32::blocked:abcd:f0 dst_f32:p:blocked:aBcd8b:f0,,,1x1x1x1,0.000976562
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_f32::blocked:abcd:f0 dst_f32:p:blocked:aBcd8b:f0,,,8x15x5x7,0.390137
onednn_verbose,exec,cpu,binary,jit:uni,undef,src_f32:p:blocked:aBcd8b:f0 src_f32:p:blocked:aBcd8b:f0 dst_f32:p:blocked:aBcd8b:f0,attr-scales:src0:0 ,alg:binary_add,8x15x5x7:1x1x1x1,0.10791
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_f32:p:blocked:aBcd8b:f0 dst_f32::blocked:abcd:f0,,,8x15x5x7,0.00488281
[ 280][DST][0:8:0:0] exp_f32:     -4.6875 exp:     -4.6875 got:         2.5 diff:  7.1875 rdiff: 1.53333
[ 281][DST][0:8:0:1] exp_f32:    -5.46875 exp:    -5.46875 got:       -3.75 diff: 1.71875 rdiff:0.314286
[ 282][DST][0:8:0:2] exp_f32:       -6.25 exp:       -6.25 got:         -10 diff:    3.75 rdiff:     0.6
[ 283][DST][0:8:0:3] exp_f32:      -4.375 exp:      -4.375 got:           5 diff:   9.375 rdiff: 2.14286
[ 284][DST][0:8:0:4] exp_f32:    -5.15625 exp:    -5.15625 got:       -1.25 diff: 3.90625 rdiff:0.757576
[ 285][DST][0:8:0:5] exp_f32:     -5.9375 exp:     -5.9375 got:        -7.5 diff:  1.5625 rdiff:0.263158
[ 286][DST][0:8:0:6] exp_f32:     -4.0625 exp:     -4.0625 got:         7.5 diff: 11.5625 rdiff: 2.84615
[ 287][DST][0:8:1:0] exp_f32:    -4.84375 exp:    -4.84375 got:        1.25 diff: 6.09375 rdiff: 1.25806
[ 288][DST][0:8:1:1] exp_f32:      -5.625 exp:      -5.625 got:          -5 diff:   0.625 rdiff:0.111111
0:FAILED (errors:1960 total:4200) __REPRO: --binary --stag=aBx8b:any --inplace=true --attr-scales=src:common:0.125* 8x15x5x7:1x1x1x1
tests:1 passed:0 skipped:0 mistrusted:0 unimplemented:0 invalid_arguments:0 failed:1 listed:0
total compute_ref: sum(s):0.00
onednn_verbose,info,oneDNN v3.0.0 (commit 5e237df1e9049c2e3e9102cab1785454db00cf8b)
onednn_verbose,info,cpu,runtime:OpenMP,nthr:32
onednn_verbose,info,cpu,isa:AArch64 SVE (256 bits)
onednn_verbose,info,gpu,runtime:none
onednn_verbose,info,prim_template:operation,engine,primitive,implementation,prop_kind,memory_descriptors,attributes,auxiliary,problem_desc,exec_time
onednn_verbose,exec,cpu,reorder,simple:any,undef,src_f32::blocked:a:f0 dst_f32::blocked:a:f0,,,1,0.626953
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_f32::blocked:abcd:f0 dst_f32:p:blocked:aBcd8b:f0,,,1x1x1x1,0.00195312
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_f32::blocked:abcd:f0 dst_f32:p:blocked:aBcd8b:f0,,,8x15x5x7,0.646973
onednn_verbose,exec,cpu,binary,jit:uni,undef,src_f32:p:blocked:aBcd8b:f0 src_f32:p:blocked:aBcd8b:f0 dst_f32:p:blocked:aBcd8b:f0,attr-scales:src1:0 ,alg:binary_add,8x15x5x7:1x1x1x1,0.35083
onednn_verbose,exec,cpu,reorder,jit:uni,undef,src_f32:p:blocked:aBcd8b:f0 dst_f32::blocked:abcd:f0,,,8x15x5x7,0.233887
[ 280][DST][0:8:0:0] exp_f32:        -7.5 exp:        -7.5 got:         2.5 diff:      10 rdiff: 1.33333
[ 281][DST][0:8:0:1] exp_f32:      -13.75 exp:      -13.75 got:       -3.75 diff:      10 rdiff:0.727273
[ 282][DST][0:8:0:2] exp_f32:         -20 exp:         -20 got:         -10 diff:      10 rdiff:     0.5
[ 283][DST][0:8:0:3] exp_f32:          -5 exp:          -5 got:           5 diff:      10 rdiff:       2
[ 284][DST][0:8:0:4] exp_f32:      -11.25 exp:      -11.25 got:       -1.25 diff:      10 rdiff:0.888889
[ 285][DST][0:8:0:5] exp_f32:       -17.5 exp:       -17.5 got:        -7.5 diff:      10 rdiff:0.571429
[ 286][DST][0:8:0:6] exp_f32:        -2.5 exp:        -2.5 got:         7.5 diff:      10 rdiff:       4
[ 287][DST][0:8:1:0] exp_f32:       -8.75 exp:       -8.75 got:        1.25 diff:      10 rdiff: 1.14286
[ 288][DST][0:8:1:1] exp_f32:         -15 exp:         -15 got:          -5 diff:      10 rdiff:0.666667
0:FAILED (errors:1960 total:4200) __REPRO: --binary --stag=aBx8b:any --inplace=true --attr-scales=src1:common:2* 8x15x5x7:1x1x1x1
```

</details>

# Expected behavior
As mentioned earlier, these tests used to pass before this patch https://github.com/oneapi-src/oneDNN/commit/7ef0b2ef2255d49ee5e334954ac8e0d2d3853ac5

**Comment by AhSinan** [comment]
It is worth noting that not all tests are failing on A64fx, but rather a subset of the above tests, specifically: 
```
0:FAILED (errors:1 total:1) __REPRO: --binary --stag=aBx8b:any --alg=ge --inplace=true 1x1x1:1x1x1
0:FAILED (errors:1 total:1) __REPRO: --binary --stag=aBx8b:any --alg=gt --inplace=true 1x1x1:1x1x1
0:FAILED (errors:1 total:1) __REPRO: --binary --stag=aBx8b:any --alg=le --inplace=true 1x1x1:1x1x1
0:FAILED (errors:1 total:1) __REPRO: --binary --stag=aBx8b:any --alg=lt --inplace=true 1x1x1:1x1x1
0:FAILED (errors:1 total:1) __REPRO: --binary --stag=aBx8b:any --alg=eq --inplace=true 1x1x1:1x1x1
0:FAILED (errors:1 total:1) __REPRO: --binary --stag=aBx8b:any --alg=ne --inplace=true 1x1x1:1x1x1
0:FAILED (errors:24 total:72) __REPRO: --binary --stag=aBx8b:any --alg=ge --inplace=true 2x12x3:2x12x3
0:FAILED (errors:24 total:72) __REPRO: --binary --stag=aBx8b:any --alg=gt --inplace=true 2x12x3:2x12x3
0:FAILED (errors:24 total:72) __REPRO: --binary --stag=aBx8b:any --alg=le --inplace=true 2x12x3:2x12x3
0:FAILED (errors:24 total:72) __REPRO: --binary --stag=aBx8b:any --alg=lt --inplace=true 2x12x3:2x12x3
0:FAILED (errors:24 total:72) __REPRO: --binary --stag=aBx8b:any --alg=eq --inplace=true 2x12x3:2x12x3
0:FAILED (errors:24 total:72) __REPRO: --binary --stag=aBx8b:any --alg=ne --inplace=true 2x12x3:2x12x3
0:FAILED (errors:22 total:264) __REPRO: --binary --stag=aBx8b:any --alg=ge --inplace=true 2x44x3:2x44x3
0:FAILED (errors:22 total:264) __REPRO: --binary --stag=aBx8b:any --alg=gt --inplace=true 2x44x3:2x44x3
0:FAILED (errors:24 total:264) __REPRO: --binary --stag=aBx8b:any --alg=le --inplace=true 2x44x3:2x44x3
0:FAILED (errors:24 total:264) __REPRO: --binary --stag=aBx8b:any --alg=lt --inplace=true 2x44x3:2x44x3
0:FAILED (errors:22 total:264) __REPRO: --binary --stag=aBx8b:any --alg=eq --inplace=true 2x44x3:2x44x3
0:FAILED (errors:24 total:264) __REPRO: --binary --stag=aBx8b:any --alg=ne --inplace=true 2x44x3:2x44x3
0:FAILED (errors:90 total:288) __REPRO: --binary --stag=aBx8b:any --alg=ge --inplace=true 2x12x3x4:2x12x3x4
0:FAILED (errors:90 total:288) __REPRO: --binary --stag=aBx8b:any --alg=gt --inplace=true 2x12x3x4:2x12x3x4
0:FAILED (errors:96 total:288) __REPRO: --binary --stag=aBx8b:any --alg=le --inplace=true 2x12x3x4:2x12x3x4
0:FAILED (errors:96 total:288) __REPRO: --binary --stag=aBx8b:any --alg=lt --inplace=true 2x12x3x4:2x12x3x4
0:FAILED (errors:90 total:288) __REPRO: --binary --stag=aBx8b:any --alg=eq --inplace=true 2x12x3x4:2x12x3x4
0:FAILED (errors:96 total:288) __REPRO: --binary --stag=aBx8b:any --alg=ne --inplace=true 2x12x3x4:2x12x3x4
0:FAILED (errors:90 total:1056) __REPRO: --binary --stag=aBx8b:any --alg=ge --inplace=true 2x44x3x4:2x44x3x4
0:FAILED (errors:90 total:1056) __REPRO: --binary --stag=aBx8b:any --alg=gt --inplace=true 2x44x3x4:2x44x3x4
0:FAILED (errors:96 total:1056) __REPRO: --binary --stag=aBx8b:any --alg=le --inplace=true 2x44x3x4:2x44x3x4
0:FAILED (errors:96 total:1056) __REPRO: --binary --stag=aBx8b:any --alg=lt --inplace=true 2x44x3x4:2x44x3x4
0:FAILED (errors:90 total:1056) __REPRO: --binary --stag=aBx8b:any --alg=eq --inplace=true 2x44x3x4:2x44x3x4
0:FAILED (errors:96 total:1056) __REPRO: --binary --stag=aBx8b:any --alg=ne --inplace=true 2x44x3x4:2x44x3x4
0:FAILED (errors:24 total:72) __REPRO: --binary --stag=aBx8b:any --inplace=true --attr-post-ops=ge:f32 2x12x3:2x12x3
0:FAILED (errors:24 total:264) __REPRO: --binary --stag=aBx8b:any --inplace=true --attr-post-ops=ge:f32 2x44x3:2x44x3
0:FAILED (errors:96 total:288) __REPRO: --binary --stag=aBx8b:any --inplace=true --attr-post-ops=ge:f32 2x12x3x4:2x12x3x4
0:FAILED (errors:96 total:1056) __REPRO: --binary --stag=aBx8b:any --inplace=true --attr-post-ops=ge:f32 2x44x3x4:2x44x3x4
```

**Comment by nSircombe** [comment]
Hi @kawakami-k,
These failures appeared in our downstream BenchDNN testing following https://github.com/oneapi-src/oneDNN/commit/7ef0b2ef2255d49ee5e334954ac8e0d2d3853ac5, originally running on Neoverse-V1 (SVE-256), but as @AhSinan mentioned, a number are also present in A64fx runs.

Have you seen these failures?

**Comment by dzarukin** [comment]
Just a tip with a context - that commit enabled back inplace mode for binary driver which was disabled for a year or so. My bold guess is that the implementation doesn't take into account that addresses might be same or something like that.

**Comment by kawakami-k** [comment]
@nSircombe 
At this point, I have not yet been able to confirm the failures and understand reason. I hope I will work to understand and fix the failures before the v3.1 branch cutoff.

**Comment by kawakami-k** [comment]
@AhSinan @nSircombe @dzarukin 

I made a path.
https://github.com/oneapi-src/oneDNN/pull/1559
Thanks for letting me know the failure.

**Comment by vpirogov** [comment]
Fixed by #1559.
